docs: warn that `useState` should only contain JSON-serializable content (#5994)
nuxt/framework: docs/content/3.api/1.composables/use-state.md

* **init**: A function that provides initial value for the state when not initiated. This function can also return a `Ref`.
* **T**: (typescript only) Specify the type of state

::alert{type=warning}

> Because the ref `useState` will be persisted from server to client, it is important that it does not contain anything (on server side) that cannot be serialized to JSON, such as a class or function. Once loaded on the client, it behaves as a normal ref and does not have this limitation.

Author: ```suggestion
Because the data inside `useState` will be persisted from server to client, it is important that it does not contain anything that cannot be serialized to JSON, such as a class or function.
```

> Because the ref `useState` creates will be persisted from server to client, it is important that it does not contain anything (on server side) that cannot be serialized to JSON, such as a class or function. Once loaded on the client, it behaves as a normal ref and does not have this limitation.

Reviewer: ```suggestion
Because the ref `useState` will be persisted from server to client, it is important that it does not contain anything (on server side) that cannot be serialized to JSON, such as a class or function. Once loaded on the client, it behaves as a normal ref and does not have this limitation.
```

Author: I think this is a rare occasion when I disagree with Damian. I think we either need to remove ref or insert creates as it was.

Reviewer: Removal of the word 'creates' was a pure assumption, as with it I couldn't grasp the sense of the sentence and thought it was left unintentionally 😄
